src/analysisd/format/json_extended.c

```diff
@@ -194,6 +194,47 @@ void W_JSON_ParseGroups(cJSON* root, const Eventinfo* lf)
         }
         token = strtok(0, delim);
     }
+
+    //Add SCA compliance groups
+    cJSON *data = cJSON_GetObjectItem(root,"data");
+    if(data){
+        cJSON *sca = cJSON_GetObjectItem(data,"sca");
+        if(sca){
+            cJSON *check = cJSON_GetObjectItem(sca,"check");
+            if(check){
+                cJSON *compliances = cJSON_GetObjectItem(check,"compliance");
+                cJSON *compliance;
+                cJSON_ArrayForEach(compliance,compliances){
+                    add_SCA_groups(rule, compliance->string, compliance->valuestring);
+                }
+            }
+        }
+    }
+}
+
+void add_SCA_groups(cJSON *rule, char* compliance, char* value){
+    char *aux;
+    os_strdup(value, aux);
+    cJSON *group = cJSON_GetObjectItem(rule, compliance);
+    if(!group){
+        group = cJSON_CreateArray();
+        cJSON_AddItemToObject(rule, compliance, group);
+    }
+    char *token;
+    char *state;
+    token = strtok_r(aux, ",", &state);
+    cJSON_AddItemToArray(group, cJSON_CreateString(token));
+    if(strlen(state) > 0){
+        if(state[0] == ' ')
+            ++state;
+        while ((token = strtok_r(NULL,",", &state))){
+            cJSON_AddItemToArray(group, cJSON_CreateString(token));
+            if(strlen(state) > 0 && state[0] == ' ')
```

Comment:
Whitespaces at the end of the token are not being trimmed as well. Why don't we use an external function to remove the whitespaces? Maybe this one could work for us making it shared:
https://github.com/wazuh/wazuh/blob/090986e4bebbfc737c551d6b4d9485acfb9d1a5c/src/addagent/manage_keys.c#L23